src/libraries/Microsoft.PowerFx.Core/Public/Config/PowerFxConfig.cs

```diff
@@ -51,6 +56,22 @@ internal static PowerFxConfig BuildWithEnumStore(CultureInfo cultureInfo, EnumSt
             return new PowerFxConfig(cultureInfo, enumStore);
         }
 
+        internal static PowerFxConfig BuildWithEnumStore(CultureInfo cultureInfo, EnumStore enumStore, IEnumerable<TexlFunction> coreFunctions)
```

Comment:
Is there a reason we can't have the core functions and extra functions included at the same time in this constructor? Why do we ever allow people to add extra functions after the object has been constructed? Don't we want to know the entire function library up-front so that we can add the required enums to the enum store?

Reply:
"Why do we ever allow people to add extra functions after the object has been constructed" this is the common pattern.  
Config is a builder object, and then passed to Engine's ctor.  Essentially, config has enough knobs on it that passing everything at once into the ctor is problematic (and would just encourage another builder pattern to initialize the config bject). 

The pattern is:

config = new Config()
config.Add(....) // other mutations here. 
engine = new Engine(config);

Reply:
But the user should know everything that they want up-front before they construct an interpreter, right? For example, let's say that you mutate a config on another thread while a binding is running. It should not affect the behavior of the binding right? For example, it should not make a "symbol not found" error stop happening halfway through type checking.

Reply:
You can't mutate the function after it's passed to an Engine() constructor, see the Check guard on all the mutating functions, so that shouldn't be possible.

Reply:
I don't like this pattern of locked/unlocked instance. It's not type safe. Why did we do it this way?

Reply:
In object-oriented land, the pattern is to use some sort of "builder" and then call .Build() to finalize and build an instance that is locked. Then only the final type is used, so that it is a type-safe operation to use an instance that is locked.

Reply:
Is that just  this:

```
config = new Config()
config.Add(....) // other mutations here.
engine = config.Build(); 
```

instead of:
```
config = new Config()
config.Add(....) // other mutations here.
engine = new Engine(config);
```

Reply:
More like this:

config = new Config()
config.Add(....) // other mutations here.
engine = new Engine(config.Build());